src/main/java/io/synadia/flink/v0/emitter/NatsRecordEmitter.java

```diff
@@ -24,8 +28,14 @@ public void emitRecord(Message element,
                            NatsSubjectSplitState splitState)
             throws Exception {
 
-        // Deserialize the message and send it to output.
-        output.collect(payloadDeserializer.getObject(splitState.getSplit().getSubject(), element.getData(), element.getHeaders()));
+        try {
+            // Deserialize the message and send it to output.
+            output.collect(payloadDeserializer.getObject(splitState.getSplit().getSubject(), element.getData(), element.getHeaders(), element.getReplyTo()));
+        } catch (Exception e) {
+            LOG.error("Failed to deserialize message", e);
+            throw new FlinkRuntimeException("Failed to deserialize message", e);
+        }
+
```

Comment:
I don't think you need to catch and rethow since the emitter interface already has `throws Exception` so 1. you would be only collecting runtime exceptions anyway, and 2) callers already handle the fact that the method throws a check Exception and will already handle it. At most I might log and re-throw, but I definitely would not change the exception type.